modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -1755,6 +1755,20 @@ def applyier(df):
     eval_general(modin_df, pandas_df, applyier)
 
 
+def test_setitem_on_empty_df_4407():
+    data = {}
+    index = pd.date_range(end="1/1/2018", periods=0, freq="D")
+    column = pd.date_range(end="1/1/2018", periods=1, freq="h")[0]
+    modin_df = pd.DataFrame(data, columns=index)
+    pandas_df = pandas.DataFrame(data, columns=index)
+
+    modin_df[column] = pd.Series([1])
+    pandas_df[column] = pandas.Series([1])
```

Comment:
why don't we explicitly test `.insert()` here? I understand that `.__setitem__` just forwards the call to `.insert()` internally, but it may change in the future
```python
>>> df = pandas.DataFrame(columns=pandas.Index([], name="some meta"))
>>> df.columns
Index([], dtype='object', name='some meta') # initial metadata
>>> df.insert(loc=0, column="name", value=[1, 2, 3])
>>> df.columns
Index(['name'], dtype='object', name='some meta') # metadata is still in place
>>>
>>> # 'freq' case
>>> df = pandas.DataFrame(columns=pandas.date_range(end="1/1/2018", periods=0, freq="D"))
>>> df.columns
DatetimeIndex([], dtype='datetime64[ns]', freq='D') # initial metadata
>>> df.insert(loc=0, column=pd.date_range(end="1/1/2018", periods=1, freq="h")[0], value=[1, 2, 3])
>>> df.columns
DatetimeIndex(['2018-01-01'], dtype='datetime64[ns]', freq='D') # metadata is still in place
```

Reply:
`insert` on empty dataframes will default to pandas, in which case `__setitem__` will not be tested.

Reply:
BTW I can add `insert` check into this test if needed.

Reply:
> `insert` on empty dataframes will default to pandas, in which case `__setitem__` will not be tested.

Oh, okay, it's fine then. Let's leave it as it is

Reply:
@naren-ponder we'll need to revisit this after we put in the empty DF PR!